causalpy/pymc_experiments.py

```diff
@@ -361,6 +360,30 @@ def __init__(
             if "post_treatment" in label and self.group_variable_name in label:
                 self.causal_impact = self.idata.posterior["beta"].isel({"coeffs": i})
 
+    def _input_validation(self):
+        """Validate the input data for correctness"""
+        assert (
```

Comment:
Here we could also have custom exceptions as above.

Reply:
done :)